File: DailyChallenge/2020/March/8.java
//https://leetcode.com/explore/challenge/card/march-leetcoding-challenge-2021/589/week-2-march-8th-march-14th/3665/

class Solution {
    public int removePalindromeSub(String s) {
        // string empty
        if (s.length() == 0)
            return 0;
        int left = 0;
        int right = s.length() - 1;
        while (left < right) {
            // palindrome condition
            if (s.charAt(left) == s.charAt(right)) {
                left += 1;
                right -= 1;
            }
            else
                return 2;
        }
        return 1;
    }
}

File: DailyChallenge/2020/April/8.cpp
// https://leetcode.com/explore/challenge/card/april-leetcoding-challenge-2021/594/week-2-april-8th-april-14th/3701/

class Solution {
public:
    unordered_map<char, string> M = {
        {'0', ""},
        {'1', ""},
        {'2', "abc"},
        {'3', "def"},
        {'4', "ghi"},
        {'5', "jkl"},
        {'6', "mno"},
        {'7', "pqrs"},
        {'8', "tuv"},
        {'9', "wxyz"}
    };

    string digits;
    int n;
    string s;
    vector<string> v;

    void next(int i) {
        if (i == n) return v.push_back(s);
        for (auto c : M[digits[i]]) { s[i] = c; next(i+1); }
    }

    vector<string> letterCombinations(string digits) {
        n = digits.size();
        this->digits = digits;
        s.resize(n);

        if (n > 0) next(0);
        return v;
    }
};
